tests/MongoDB.Driver.Core.Tests/Core/Servers/ServerFactoryTests.cs

```diff
@@ -44,6 +44,11 @@ public ServerFactoryTests()
             _connectionPoolFactory = new Mock<IConnectionPoolFactory>().Object;
             _endPoint = new DnsEndPoint("localhost", 27017);
             _serverMonitorFactory = new Mock<IServerMonitorFactory>().Object;
+            var mockServerMonitor = new Mock<IServerMonitor>();
+            mockServerMonitor.Setup(m => m.Description).Returns(new ServerDescription(new ServerId(_clusterId, _endPoint), _endPoint));
+            var mockServerMonitorFactory = new Mock<IServerMonitorFactory>();
+            mockServerMonitorFactory.Setup(f => f.Create(It.IsAny<ServerId>(), _endPoint)).Returns(mockServerMonitor.Object);
+            _serverMonitorFactory = mockServerMonitorFactory.Object;
```

Comment:
`CreateServer_should_return_Server` fails without these lines because `Server`'s constructor now needs to access the ServerMonitor's description

Reply:
Server no longer needs to access the monitor's description but these lines do no harm: they could help with future tests. That being said, I'm happy to strike them if reviewers feel that this is unneeded.